### PR TITLE
fix(storage): add error handling to setting or getting storage

### PR DESCRIPTION
**Pull request by anthony-liddle**
## Goal

Addressing errors captured by Sentry.

The storage errors seem to be the biggest culprit with the authentication issues experienced by some users. If Chrome can't `get` or `set` items to the local storage, there is no way to save a user's auth status. 

to be merged into https://github.com/Pocket/extension-save-to-pocket/pull/276

image

## Todos:
- [x] Bump package version
- [x] Add error handling to getting and setting `chrome.storage`
- [x] Update message listener in background to return a promise
- [x] Remove un-used check for system theme
- [x] Adding ability to display error messaging to user
- [x] Adding generic error message to localization files

### IO error: ... FILE_ERROR_NO_SPACE (ChromeMethodBFE: 3::WritableFileAppend::8)
This is a Chrome level error and is related to a user's available disc space when trying to write to `chrome.storage`. There isn't any _real_ documentation on this error provided by Google. In my research I came across another open-source extension where users experienced this issue and it was solved by clearing hard-drive space and rebooting the browser. Not ideal. I've added error handling to our helper function that sets the storage, as well as error logging into Sentry

-----------------------------------

### IO error: .../LOCK: File currently in use. (ChromeMethodBFE: 15::LockFile::2)
This another Chrome level error is caused by Chrome leaving old or "zombie" processes running. Here's a thread that discusses these processes in regards to a similar (but seemingly related) issue: https://bugs.chromium.org/p/chromium/issues/detail?id=490915

Similar to the no space error, there is no documentation provided by Google for this issue. However research surfaced two other extensions with this issue that were fixed by restarting the browser, or restarting the computer. I've added error logging into Sentry when this action fails.

-----------------------------------

### The message port closed before a response was received.
When you add a listener on browser message events you always need to return a promise. Documentation:
- https://developer.chrome.com/docs/extensions/reference/runtime/#event-onMessage
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage
- https://github.com/mozilla/webextension-polyfill/issues/130

### CODE REVIEW

File: src/pages/background/index.js [hunk 1]
```diff
@@ -57,34 +57,37 @@ chrome.runtime.onMessage.addListener(function (message, sender) {
...
     case OPEN_OPTIONS:
       handle.openOptionsPage()
-      return
+      break
     default:
-      return
+      break
   }
+
+  return Promise.resolve(`Message received: ${type}`)
```
Comment (anthony-liddle) [inline src/pages/background/index.js:92]: Returning this Promise prevents the `The message port closed before a response was received` error

File: src/pages/injector/content.js [hunk 2]
```diff
@@ -11,18 +11,6 @@ const initialize = function () {
...
-  const mediaQueryList = window.matchMedia('(prefers-color-scheme: dark)')
-  const handleDarkModeChange = (mql) => {
-    const darkMode = mql.matches
-    chrome.runtime.sendMessage({
-      type: COLOR_MODE_CHANGE,
-      payload: { darkMode },
-    })
-  }
-  handleDarkModeChange(mediaQueryList)
-  mediaQueryList.addEventListener('change', handleDarkModeChange)
```
Comment (anthony-liddle) [inline src/pages/injector/content.js:25]: This check wasn't actually hooked up correctly, so it wasn't working as expected. We can re-add later if we feel it's important, but for now it's gotta go

File: src/common/interface.js [hunk 3]
```diff
@@ -56,11 +58,34 @@ export function inactiveIcon() {
...
+  })
+}
+
+function handleSettingError(message) {
+  Sentry.withScope((scope) => {
+    scope.setFingerprint('Storage Error')
+    Sentry.captureMessage(`Storage Error: ${message}`)
+  })
+
+  console.error(message)
```
Comment (anthony-liddle) [inline src/common/interface.js:90]: Logging the message here as it's helpful context if the user contacts support.

File: src/pages/background/userActions.js [hunk 4]
```diff
@@ -66,16 +66,16 @@ export function contextClick(info, tab) {
...
   chrome.tabs.sendMessage(tabId, { action: SAVE_TO_POCKET_REQUEST })
 
-  const url = linkUrl || pageUrl
-
   try {
+    // Are we authed?
+    const access_token = await getSetting('access_token')
+    if (!access_token) return logIn({ linkUrl, pageUrl, title, tabId })
+
+    const url = linkUrl || pageUrl
```
Comment (anthony-liddle) [inline src/pages/background/userActions.js:77]: These 3 lines came from above. The only difference here is that both checks to local storage are now wrapped with try, and we open the panel with the `SAVE_TO_POCKET_REQUEST` action immediately.

The visual difference is the panel will open with the "Saving..." text, and will remain until the user has completed the login process.

File: src/components/error-message/error-message.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { css } from 'linaria'
+
+const errorMessageStyles = css`
+  color: red;
```
Comment (wtfluckey) [inline src/components/error-message/error-message.js:5]: Should we use one of the color variables here?

**Review by wtfluckey — APPROVED** [review]

File: src/components/error-message/error-message.js (hunk 5 above)
Comment (anthony-liddle) [reply src/components/error-message/error-message.js]: Good call! My placeholder got away from me.

**Review by collectedmind — APPROVED** [review]

**Pull request merged**